src/router.ts

```diff
@@ -0,0 +1,26 @@
+const express = require('express');
+import * as glob from 'glob'
+import logger from './util/logger';
+import * as path from 'path';
+//key index issue with types:so any
+const router:any = express.Router(); 
+
+const routes_list:Array<[string,string,Function]> = [];
+logger.info('Importing routes...')
+
+glob.sync( 'src/routes/**/*.ts' ).forEach((file:string)=>{
+    routes_list.push(require(path.resolve(file)).default);
+});
+  
+for(const [path,method,fn] of routes_list) {
+    logger.info(`Adding route: ${path}`);
+    try{
+        router[(method).toLowerCase()](path,fn);
+    }catch(e){
+        logger.warn(e);
+    }   
+}
```

Comment:
Genuinely curious: is this better/preferential compared to making sub-routers and registering the routes manually?

So long as it's strongly typed, it shouldn't cause any problems, but I've never seen routes registered through filepath loading before.

Reply:
it is odd, I agree.

I was writing a router/controller version before reedthorngag started this pattern. I have used this before (the globbing idea) in a template repo I use for quick prototyping. the router/controller version has a bit more boilerplate, so I switched to this.

Reply:
I just like it because of how simple and hard to break it is, you don't need to remember to add stuff to files or update it when you change it, you can just add a file anywhere it src/routes, including in subfolders and it will handle it perfectly without any extra effort, I personally dont like having to go to other files and register it there, just my preference, I try to keep everything self contained if possible so you only need to change things in one place